classes/fields/file.php

```diff
@@ -1141,8 +1172,60 @@ public function admin_ajax_upload() {
 			$custom_handler = apply_filters( 'pods_upload_handle', null, 'Filedata', $params->post_id, $params, $field );
 
 			if ( null === $custom_handler ) {
+
+				// Start custom directory.
+				$upload_dir = pods_v( $field['type'] . '_upload_dir', $field['options'], 'wp' );
+
+				if ( 'wp' !== $upload_dir ) {
+					$custom_dir = pods_v( $field['type'] . '_upload_dir_custom', $field['options'], '' );
+
+					$context_pod = null;
+					if ( $params->post_id ) {
+						$post = get_post( $params->post_id );
+						if ( $post ) {
+							$post_pod = pods( $post->post_type, $post->ID );
+							if ( $post_pod->exists() ) {
+								$context_pod = $post_pod;
+							}
+						}
+					}
+
+					/**
+					 * Filter the custom upload directory Pod context.
+					 * @since 2.7.28
+					 * @param Pods   $context_pod
+					 * @param array  $params
+					 * @param array  $field
+					 * @param array  $pod
```

Comment:
```suggestion
					 * Filter the custom upload directory Pod context.
					 *
					 * @since 2.7.28
					 *
					 * @param Pods  $context_pod The Pods object of the associated pod for the post type.
					 * @param array $params      The POSTed parameters for the request.
					 * @param array $field       The field configuration associated to the upload field.
					 * @param array $pod         The pod configuration associated to the upload field.
```